public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -501,6 +521,60 @@ export class GraphiteDatasource
     }
   }
 
+  /**
+   * Search for metrics matching giving pattern using /metrics/render endpoint.
+   * It will return all possible values and parse them based on queryType.
+   * For example:
+   *
+   * queryType: GraphiteQueryType.Value
+   * query: groupByNode(movingAverage(apps.country.IE.counters.requests.count, 10), 2, 'sum')
+   * result: 239.4, 233.4, 230.8, 230.4, 233.9, 238, 239.8, 236.8, 235.8
+   *
+   * queryType: GraphiteQueryType.MetricName
+   * query: apps.backend.*.counters.requests.count
+   * result:
+   *     apps.backend.backend_01.counters.requests.count
+   *     apps.backend.backend_02.counters.requests.count
+   *     apps.backend.backend_03.counters.requests.count
+   *     apps.backend.backend_04.counters.requests.count
+   */
+  private async requestMetricRender(
+    queryObject: GraphiteQuery,
+    requestId: string,
+    range: TimeRange
+  ): Promise<MetricFindValue[]> {
+    const queryReq: DataQueryRequest<GraphiteQuery> = {
```

Comment:
Should this be the same for all queries? I mean if we want metric names isn't there a way to make the query more lightweight or isn't there an endpoint specific to metric names like in prometheus?

Reply:
Thanks for the suggestion @aocenas. So I took a step back and explored what we have already and saw that `metrics/expand` endpoint is actually doing what we need. So I used it. Please see: https://github.com/grafana/grafana/pull/52341/commits/413129fbb27fee9d1dbc644f96a3faf575f30338